runtime/src/snapshot_utils.rs

```diff
@@ -843,8 +843,19 @@ pub fn create_accounts_run_and_snapshot_dirs(
 ) -> std::io::Result<(PathBuf, PathBuf)> {
     let run_path = account_dir.as_ref().join("run");
     let snapshot_path = account_dir.as_ref().join("snapshot");
-    fs::create_dir_all(&run_path)?;
-    fs::create_dir_all(&snapshot_path)?;
+    if fs::metadata(&run_path).is_err() || fs::metadata(&snapshot_path).is_err() {
+        // If the "run/" or "snapshot" sub directories do not exist, the directory may be from
+        // an older version for which the appendvec files are at this directory.  Clean up
+        // them first.
+        // This will be done only once when transitioning from an old image without run directory
+        // to this new version using run and snapshot directories.
+        // The run/ content cleanup will be done at a later point.  The snapshot/ content persists
+        // accross the process boot, and will be purged by the account_background_service.
+        move_and_async_delete_path(account_dir.as_ref());
```

Comment:
Looking at the metrics, it is fine.  Deleting (even asynchronously) happens much faster than unpacking.

Reply:
OK.  will change to synchronous.  need to run the tests for that.

Reply:
it actually gets more complicated.  
sync rmdir would fail on /mnt/accounts/, so i need to make snapshot_utils.rs::delete_contents_of_path public.  to do that cleanly, it is better to expose a function similar to move_and_async_delete_path, but in sync version, or rename that function to take a sync/async flag.

Not sure it is worth the effort for this very low chance.

Reply:
ok, i don't have to make delete_contents_of_path public.  i will just call it when remove_dir_all fails.